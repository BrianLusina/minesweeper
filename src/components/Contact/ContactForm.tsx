import Button from '@components/ui/Button';
import { isValidEmail } from '@utils';
import { FunctionComponent, useState, FormEvent } from 'react';

const ContactForm: FunctionComponent = () => {
  const [name, setName] = useState<string>('');
  const [email, setEmail] = useState<string>('');
  const [message, setMessage] = useState<string>('');

  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
  };

  const isValid = (): boolean => {
    return isValidEmail(email) && name !== '' && message !== '';
  };

  const onSubmit = () => {
    //
  };

  return (
    <>
      <form method="post" onSubmit={handleSubmit}>
        <div className="row gtr-uniform gtr-50">
          <div className="col-6 col-12-xsmall">
            <input
              type="text"
              name="name"
              id="name"
              placeholder="Name"
              value={name}
              onChange={(e) => setName(e.target.value)}
            />
          </div>
          <div className="col-6 col-12-xsmall">
            <input
              type="email"
              name="email"
              id="email"
              placeholder="Email"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
            />
          </div>
          <div className="col-12">
            <textarea
              name="message"
              id="message"
              placeholder="Message"
              rows={4}
              value={message}
              onChange={(e) => setMessage(e.target.value)}
            />
          </div>
        </div>
      </form>
      <ul className="actions">
        <li>
          <Button type="submit" onClick={onSubmit} text="Send message" disabled={!isValid()} />
        </li>
      </ul>
    </>
  );
};

export default ContactForm;
