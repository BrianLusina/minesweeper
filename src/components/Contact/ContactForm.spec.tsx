import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ContactForm from './ContactForm';

describe('ContactForm', () => {
  it('should render', () => {
    const { container } = render(<ContactForm />);

    expect(container).toMatchSnapshot();
  });

  it('should validate form and not allow button click with invalid fields', () => {
    render(<ContactForm />);
    const nameText = 'John Doe';
    const email = 'invalid';
    const message = 'message';

    const nameInput = screen.getByPlaceholderText('Name');
    const emailInput = screen.getByPlaceholderText('Email');
    const messageInput = screen.getByPlaceholderText('Message');
    const button = screen.getByRole('button');

    userEvent.type(nameInput, nameText);
    userEvent.type(emailInput, email);
    userEvent.type(messageInput, message);

    userEvent.click(button);

    // no API call is made to submit form
  });
});
