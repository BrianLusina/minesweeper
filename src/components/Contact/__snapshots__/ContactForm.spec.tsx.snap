// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should render 1`] = `
<div>
  <form
    method="post"
  >
    <div
      class="row gtr-uniform gtr-50"
    >
      <div
        class="col-6 col-12-xsmall"
      >
        <input
          id="name"
          name="name"
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="col-6 col-12-xsmall"
      >
        <input
          id="email"
          name="email"
          placeholder="Email"
          type="email"
          value=""
        />
      </div>
      <div
        class="col-12"
      >
        <textarea
          id="message"
          name="message"
          placeholder="Message"
          rows="4"
        />
      </div>
    </div>
  </form>
  <ul
    class="actions"
  >
    <li>
      <button
        class="button disabled"
        disabled=""
        type="submit"
      >
        Send message
      </button>
    </li>
  </ul>
</div>
`;
